compiler/rustc_target/src/asm/sparc.rs

```diff
@@ -0,0 +1,135 @@
+use std::fmt;
+
+use rustc_data_structures::fx::FxIndexSet;
+use rustc_span::{Symbol, sym};
+
+use super::{InlineAsmArch, InlineAsmType, ModifierInfo};
+use crate::spec::{RelocModel, Target};
+
+def_reg_class! {
+    Sparc SparcInlineAsmRegClass {
+        reg,
+        yreg,
+    }
+}
+
+impl SparcInlineAsmRegClass {
+    pub fn valid_modifiers(self, _arch: super::InlineAsmArch) -> &'static [char] {
+        &[]
+    }
+
+    pub fn suggest_class(self, _arch: InlineAsmArch, _ty: InlineAsmType) -> Option<Self> {
+        None
+    }
+
+    pub fn suggest_modifier(
+        self,
+        _arch: InlineAsmArch,
+        _ty: InlineAsmType,
+    ) -> Option<ModifierInfo> {
+        None
+    }
+
+    pub fn default_modifier(self, _arch: InlineAsmArch) -> Option<ModifierInfo> {
+        None
+    }
+
+    pub fn supported_types(
+        self,
+        arch: InlineAsmArch,
+    ) -> &'static [(InlineAsmType, Option<Symbol>)] {
+        match self {
+            Self::reg => {
+                // FIXME: i64 is ok for g*/o* registers on SPARC-V8+ ("h" constraint in GCC)
```

Comment:
This is better expressed as `"v9": I64` in the macro. The feature name can later be changed to `v8plus`.

Reply:
That constraint is not yet supported in LLVM, so I have updated the FIXME comment to use the way you mention.

https://github.com/rust-lang/rust/blob/241f82ad915b167992ec9d3bb729f095a7829424/compiler/rustc_target/src/asm/sparc.rs#L44-L49